### PR TITLE
[DOCFIX] Reorganize client doc pages

### PR DESCRIPTION
**Pull request by gpang**
@apc999 @aaudiber

### CODE REVIEW

**Comment by apc999** [comment]
thanks for the change. do we also move the Javadoc and RestAPI doc to `resources`?

File: docs/en/Clients-Hadoop.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+{:toc}
+
+Alluxio provides access to data through a filesystem interface. Files in Alluxio offer write-once
+semantics: they become immutable after they have been written in their entirety and cannot be read
+before being completed. Alluxio provides two different Filesystem APIs, a native API and a Hadoop
+compatible API. The native API provides better performance, while the Hadoop compatible API gives
+users the flexibility of leveraging Alluxio without having to modify existing code written using
+Hadoop's API.
+
+Alluxio has a wrapper of the native client which provides the Hadoop compatible `FileSystem`
```
Comment (apc999) [inline docs/en/Clients-Hadoop.md]: put a pointer of native client?

File: docs/en/Clients-Rest.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+---
+layout: global
+title: REST API
+nickname: REST API
+group: Clients
+priority: 3
+---
+
+For portability with other languages, the Alluxio native API is also accessible via an HTTP proxy in
```
Comment (apc999) [inline docs/en/Clients-Rest.md]: link Alluxio native API

File: docs/en/Clients-Rest.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+the form of a REST API.
+
+The REST API documentation is generated as part of Alluxio build and accessible through
```
Comment (apc999) [inline docs/en/Clients-Rest.md]: link The REST API documentation?

File: docs/en/Clients-Rest.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+`${ALLUXIO_HOME}/core/server/proxy/target/miredot/index.html`. In particular, the `paths`
```
Comment (apc999) [inline docs/en/Clients-Rest.md]: not sure if we need to go down to this level of details, especially we can put the rest API link here

File: docs/en/Clients-Hadoop.md (hunk 1 above)
Comment (gpang) [inline docs/en/Clients-Hadoop.md]: done

File: docs/en/Clients-Rest.md (hunk 2 above)
Comment (gpang) [inline docs/en/Clients-Rest.md]: done

File: docs/en/Clients-Rest.md (hunk 3 above)
Comment (gpang) [inline docs/en/Clients-Rest.md]: done

File: docs/en/Clients-Rest.md (hunk 4 above)
Comment (gpang) [inline docs/en/Clients-Rest.md]: done

File: docs/en/Clients-Hadoop.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+compatible API. The native API provides better performance, while the Hadoop compatible API gives
+users the flexibility of leveraging Alluxio without having to modify existing code written using
+Hadoop's API.
+
+Alluxio has a wrapper of the [native client](Clients-Java-Native.html) which provides the Hadoop
+compatible `FileSystem` interface. With this client, Hadoop file operations will be translated to
+FileSystem operations. The latest documentation for the `FileSystem` interface may be found
+[here](http://hadoop.apache.org/docs/current/api/org/apache/hadoop/fs/FileSystem.html).
+
+The Hadoop compatible interface is provided as a convenience class, allowing users to retain
```
Comment (aaudiber) [inline docs/en/Clients-Hadoop.md]: nit: retain -> reuse

File: docs/en/Clients-Hadoop.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+priority: 2
+---
+
+* Table of Contents
+{:toc}
+
+Alluxio provides access to data through a filesystem interface. Files in Alluxio offer write-once
+semantics: they become immutable after they have been written in their entirety and cannot be read
+before being completed. Alluxio provides two different Filesystem APIs, a native API and a Hadoop
+compatible API. The native API provides better performance, while the Hadoop compatible API gives
```
Comment (aaudiber) [inline docs/en/Clients-Hadoop.md]: the performance of the native API is very similar, can we remove the performance part? Same in Alluxio-Clients-Java-Native

File: docs/en/Clients-Rest.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+the REST API and the Native API is in how streams are represented. While the native API
+can use in-memory streams, the REST API decouples the stream creation and access (see the
+`create` and `open` REST API methods and the `streams` resource endpoints for details).
+
+The HTTP proxy is a standalone server that can be started using
+`${ALLUXIO_HOME}/bin/alluxio-start.sh proxy` and stopped using `${ALLUXIO_HOME}/bin/alluxio-stop.sh
+proxy`. By default, the REST API is available on port 39999.
+
+There are performance implications of using the HTTP proxy. In particular, using the proxy requires
+an extra hop. For optimal performance, it is recommended to run the proxy server an Alluxio worker
```
Comment (aaudiber) [inline docs/en/Clients-Rest.md]: proxy server an Alluxio worker -> proxy server and an Alluxio worker

File: docs/en/Clients-Hadoop.md (hunk 6 above)
Comment (gpang) [inline docs/en/Clients-Hadoop.md]: done

File: docs/en/Clients-Hadoop.md (hunk 5 above)
Comment (gpang) [inline docs/en/Clients-Hadoop.md]: done

File: docs/en/Clients-Rest.md (hunk 7 above)
Comment (gpang) [inline docs/en/Clients-Rest.md]: done

**Comment by aaudiber** [comment]
Are we still planning to move the javadoc and restdoc to the resources section?

**Pull request merged**